pkg/parse/reconciler.go

```diff
@@ -83,23 +63,14 @@ func (p *reconciler) Options() *ReconcilerOptions {
 	return p.options
 }
 
-// SyncStatusClient returns the SyncStatusClient used by this reconciler.
-func (p *reconciler) SyncStatusClient() SyncStatusClient {
-	return p.syncStatusClient
-}
-
-// ReconcilerState returns the current state of the reconciler.
-func (p *reconciler) Parser() Parser {
-	return p.parser
-}
-
 // ReconcilerState returns the current state of the reconciler.
 func (p *reconciler) ReconcilerState() *ReconcilerState {
 	return p.reconcilerState
 }
 
-// NewRootRunner creates a new runnable parser for parsing a Root repository.
-func NewRootRunner(recOpts *ReconcilerOptions, parseOpts *RootOptions) Reconciler {
+// NewClusterReconciler creates a new reconciler for reconciling cluster-scoped
+// and namespace-scoped resources configured with a RootSync.
+func NewClusterReconciler(recOpts *ReconcilerOptions, parseOpts *RootOptions) Reconciler {
```

Comment:
nit: I think `NewRootSyncReconciler` and `NewRepoSyncReconciler` may be more intuitive. `Cluster` and `Namespace` seem not as precise as they could be

Reply:
I considered that. I'm not against it.

It seems to me that it's primarily reconciling the cluster objects (cluster-scoped and/or namespace-scoped) with the source objects, but as a controller it's also reconciling the RSync status with the spec...

Reply:
But if that wasn't immediately obvious from the name, then I'm ok with your suggestion, which should be pretty clear.

Reply:
I wish we could rename the resources themselves to be more clear, but that ship sailed before my time.

Reply:
Renamed.